org.adempiere.base/src/org/compiere/model/MAllocationHdr.java

```diff
@@ -327,17 +327,37 @@ protected boolean beforeDelete ()
 
 		//	Unlink
 		getLines(true);
-		if (!updateBP(true)) 
+		String prevStatus = getDocStatus();
+		setDocStatus(STATUS_Reversed);
+		if (!updateBP())

```

Comment:
why the need to call updateBP() here if setTotalOpenBalance is call for each bp in afterDelete() ?